presto-main/src/main/java/com/facebook/presto/sql/relational/optimizer/ExpressionOptimizer.java

```diff
@@ -235,36 +234,39 @@ private CallExpression rewriteCast(CallExpression call)
                     TypeSignature returnType = call.getSignature().getReturnType();
                     if (returnType.getBase().equals(ARRAY)) {
                         return call(
-                                internalScalarFunction(
-                                        JSON_STRING_TO_ARRAY_NAME,
-                                        returnType,
-                                        ImmutableList.of(parseTypeSignature(VARCHAR))),
+                                functionManager.lookupCast(
```

Comment:
This is great :) Don't understand why we wrote this as a fixed internalScalarFunction before ...